install.sh

```diff
@@ -298,7 +298,10 @@ fi
 if [[ -z "${HOMEBREW_ON_LINUX-}" ]]; then
  have_sudo_access
 else
-  if [[ -n "${CI-}" ]] || [[ -w "$HOMEBREW_PREFIX_DEFAULT" ]] || [[ -w "/home/linuxbrew" ]] || [[ -w "/home" ]]; then
+  if [[ -n "${NONINTERACTIVE-}" ]] ||
+     [[ -w "$HOMEBREW_PREFIX_DEFAULT" ]] ||
+     [[ -w "/home/linuxbrew" ]] ||
+     [[ -w "/home" ]]; then
     HOMEBREW_PREFIX="$HOMEBREW_PREFIX_DEFAULT"
```

Comment:
Using NONINTERACTIVE here makes it now impossible to install linuxbrew non-interactively in the home folder. Is that really what we want?

With this change, if you are non-interactively installing linuxbrew and don't have elevated permissions, the script will fail. Before this change, it would just install in the home folder with no problem. (You can even force that behavior with `export SUDO_ASKPASS=false`. 

If you just want to get rid of the CI variable, I would just completely remove this check. It doesn't makes sense to force install in the `/home/linuxbrew` if run NONINTERACTIVE, the script should decide that itself. 

I think the reason the check used to be here is that it was very likely that the script would block on the else block after this. So, if on CI, you didn't want to run that piece of code. But now we are more clever in the else block and won't block anymore. But since people weren't bothered setting the CI variable (undocumented?), still a lot of CI pipelines broke when we made the else block more interactive.

Reply:
> Using NONINTERACTIVE here makes it now impossible to install linuxbrew non-interactively in the home folder. Is that really what we want?

Yes, we should non-interactively always install in the default, supported location if the user isn't being shown the `recommended` message.

> But since people weren't bothered setting the CI variable (undocumented?), still a lot of CI pipelines broke when we made the else block more interactive.

Most CI pipelines do set this variable, though?

Reply:
>Yes, we should non-interactively always install in the default, supported location if the user isn't being shown the recommended message.

That is fine in that case!

>Most CI pipelines do set this variable, though?

I didn't knew that. Anyway, if you dig into the comments of #341 apparently not all do. For example, this person found that it installed in the home folder, so he did not had the CI variable set in that case. https://github.com/Homebrew/install/pull/341#issuecomment-733915458